### ISSUE TITLE
Migrate on QML

### ISSUE THREAD

**Issue by Diadlo** [qTox/planning#2]
I'm not sure, that it's good idea. But we can start rewriting part of widgets into QML
http://doc.qt.io/qt-4.8/qml-integration.html

**Comment by Diadlo** [comment]
@sudden6 Do you agree with this and #1?

**Comment by antis81** [comment]
I would strongly advise to step away from that idea of mixing widget and qml code. However, if you want to try it, you might want to have a look into my fork (HEAD commit of branch `qml`) for reference on how to do that: https://github.com/antis81/qTox/commit/017bdbc0e4d9979fdfbed12fe1c0b641614e66b2

The commit basically removes any QWidget related code - but you can easily just add the qml/qtquick module and start off with that…

**Comment by Diadlo** [comment]
@antis81 Thanks a lot
> I would strongly advise to step away from that idea of mixing widget and qml code

Why? There is my arguments for mixing:
1) I think QML -- is the future of Qt
2) QML help to build better architecture (MVVM?)
3) We can't drop support of current futures => we can't start from scratch
4) We can't maintain (test, fix, rebase) one huge PR

**Comment by sudden6** [comment]
I agree with @Diadlo, we don't have the manpower to do a hard switch all at once, so we have to do it gradually, even if it takes longer. But it means our users will get continuous support during the whole process.
